plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java

```diff
@@ -62,6 +62,12 @@ public void eclipseFormatFile(Object eclipseFormatFile) {
 	}
 
 	public void eclipseFormatFile(String eclipseVersion, Object... eclipseFormatFiles) {
+		// disambiguate the ambiguous methods
+		// see ScalaExtension.scalafmt() for a much better mechanism: https://github.com/diffplug/spotless/blob/a8ab1c1331376e57e63441417316139198c285de/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/ScalaExtension.java#L38-L65
+		if (eclipseFormatFiles.length == 0) {
+			eclipseFormatFiles = new Object[]{eclipseVersion};
+			eclipseVersion = EclipseFormatterStep.defaultVersion();
+		}
 		Project project = getProject();
```

Comment:
I can see how it would break my single-arg-and-iterable workaround, but not for my other one.

I'm not really familiar with Groovy, but if it were Java, I believe this would be syntactically correct if used with my latter workaround:
```gradle
greclipseFormatFile('mycfg1.prop', 'mycfg2.prop');
```